### PR TITLE
[SPARK-40491][SQL] Remove too old TODO for JdbcRDD

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
According to the legacy document of `JdbcRDD`, we need to expose a jdbcRDD function in `SparkContext`.

### Why are the changes needed?
The `jdbcRDD` function in `SparkContext` easy use for users.

### Does this PR introduce _any_ user-facing change?
'No'.
Just is a new API.

### How was this patch tested?
New tests.

### CODE REVIEW

**Review by HyukjinKwon — COMMENTED** [review]
Hm, why do we need this? Can't we do `spark.read.jdbc(...).rdd` or `toDS`?

**Comment by beliefer** [comment]
> Hm, why do we need this? Can't we do `spark.read.jdbc(...).rdd` or `toDS`?

I know. This PR just follows the legacy document of `JdbcRDD`. If we don't need the change, we may need update the comments.

**Comment by beliefer** [comment]
The failure test is not related to this PR.

File: core/src/main/scala/org/apache/spark/rdd/JdbcRDD.scala [hunk 1]
```diff
@@ -32,7 +32,6 @@ private[spark] class JdbcPartition(idx: Int, val lower: Long, val upper: Long) e
   override def index: Int = idx
 }
 
-// TODO: Expose a jdbcRDD function in SparkContext and mark this as semi-private
```
Comment (cloud-fan) [inline core/src/main/scala/org/apache/spark/rdd/JdbcRDD.scala:35]: This is a very old TODO and we need to revisit if this is still necessary. Since Spark SQL is the new core, I'm not sure if anyone is interested in a new API to create jdbc RDD.

Comment (beliefer) [reply core/src/main/scala/org/apache/spark/rdd/JdbcRDD.scala:35]: So, could we remove this TODO comment ?

Comment (cloud-fan) [reply core/src/main/scala/org/apache/spark/rdd/JdbcRDD.scala:35]: yea

File: core/src/main/scala/org/apache/spark/api/java/JavaSparkContext.scala [hunk 2]
```diff
@@ -182,8 +182,6 @@ class JavaSparkContext(val sc: SparkContext) extends Closeable {
   def textFile(path: String, minPartitions: Int): JavaRDD[String] =
     sc.textFile(path, minPartitions)
 
-
-
```
Comment (HyukjinKwon) [inline core/src/main/scala/org/apache/spark/api/java/JavaSparkContext.scala:186]: Can we revert this change back?

**Review by HyukjinKwon — APPROVED** [review]

File: core/src/main/scala/org/apache/spark/api/java/JavaSparkContext.scala (hunk 2 above)
Comment (beliefer) [reply core/src/main/scala/org/apache/spark/api/java/JavaSparkContext.scala:186]: It seems these blanks are useless.

Comment (beliefer) [reply core/src/main/scala/org/apache/spark/api/java/JavaSparkContext.scala:186]: OK

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by beliefer** [comment]
@HyukjinKwon @cloud-fan Thank you for all !
